src/NearestNeighbors.jl

```diff
@@ -23,7 +23,6 @@ abstract type NNTree{V <: AbstractVector,P <: PreMetric} end
 
 const NonweightedMinowskiMetric = Union{Euclidean,Chebyshev,Cityblock,Minkowski}
 const WeightedMinowskiMetric = Union{WeightedEuclidean,WeightedCityblock,WeightedMinkowski}
```

Comment:
@dkarrasch, should I keep these unions? I can make methods for each type in the `eval_pow` and `eval_end` functions instead.

Reply:
Though that sounds reasonable, because external packages won't be able to subscribe to these union types, who knows if other packages use them to restrict method signatures, so better leave them to avoid unnecessary breakage.